zulipterminal/helper.py

```diff
@@ -162,12 +162,12 @@ def _set_count_in_view(controller: Any, new_count: int,
         additionally set the current count in the model and make use of the
         same in the UI.
     """
-    stream_buttons_log = controller.view.stream_w.log
+    stream_buttons_list = controller.view.stream_w.streams_btn_list
```

Comment:
Could you discuss the need for these changes further? This seems like an interesting change in any case, since we update the log during search, so perhaps it belongs in a separate commit - but I'd like to hear your view on it.

Reply:
Since, the log is updated while searching at some point we hit an attribute error because Text wrapped with AttrMap does not contain `stream_id`. Even if the log wasn't empty this would mean message updates are only sent to a subset of streams, topics or users. Since, we only need to update the count variable in the button object we can use the `btn_list` instead.

Reply:
Hopefully this doesn't conflict with the rest of the code, since a `'list'` was replaced by another `list`

Reply:
Actually I had a question here, though this is not related to this PR. Why do we only update count for topic only when topic view is open? Is it because there are too many topics to handle at once and instead when a user toggles into the stream, we get the latest count from the server?

Reply:
There are a few different points here:
* Normally if this just fixed a bug caused by an earlier commit in this PR, it's good practice to try and place this commit earlier as a refactor (1)
* Re the AttrMap point, I'm guessing this is related to how the `log` code works; we work around this kind of thing with the pinned-streams-divider by assigning extra attributes to a simple Divider, but...
* Most importantly, it does seem that if we're in a filtered subset then the count won't update, as you say. However, I can't reproduce this situation or a fix with this code, so we may currently have another issue which limits this fix (2)

Given the separate nature of this commit I'd suggest including it (2) but moving it earlier (1). I'm definitely interested if this corresponds to some unreads edge cases, but as I say I've not been able to make a difference with this code.

We're only iterating through a sequence, so this shouldn't make a huge difference.

Re topic view, we don't have a permanent topic view indefinitely for each stream, and the left panel switches content and loads on demand from the model, from memory.

Reply:
Re unread edge cases, I have reproduced a particular edge case for negative counts which can be found in #642 as well as in [#zulip-terminal - Negative unread counts](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/Negative.20unread.20counts)